controller-server/src/main/java/com/yahoo/vespa/hosted/controller/dns/RemoveRecords.java

```diff
@@ -64,32 +55,18 @@ public Optional<RecordData> data() {
 
     @Override
     public void dispatchTo(NameService nameService) {
-        List<Record> records = new ArrayList<>();
-        if (name.isPresent() && data.isPresent()) {
-            nameService.findRecords(type, name.get())
-                       .stream()
-                       .filter(record -> {
-                           // Records to remove must match both name and data fields
-                           String dataValue = switch (record.type()) {
-                               case ALIAS -> AliasTarget.unpack(record.data()).name().value();
-                               case DIRECT -> DirectTarget.unpack(record.data()).recordData().asString();
-                               default -> record.data().asString();
-                           };
-                           return fqdn(dataValue).equals(fqdn(data.get().asString()));
-                       })
-                       .forEach(records::add);
-        } else {
-            name.ifPresent(n -> records.addAll(nameService.findRecords(type, n)));
-            data.ifPresent(d -> records.addAll(nameService.findRecords(type, d)));
-        }
-        nameService.removeRecords(records);
+        // Deletions require all records fields to match exactly, data may be incomplete even if present. To ensure
+        // completeness we search for the record(s) first
+        List<Record> completeRecords = nameService.findRecords(type, name).stream()
+                                                  .filter(record -> data.isEmpty() || matchingFqdnIn(data.get(), record))
+                                                  .toList();
+        nameService.removeRecords(completeRecords);
```

Comment:
So we just drop deletion of unmatched records? Should we log a warning when this happens?

Reply:
The logic should be the same as before this change. Not sure what you mean by unmatched here? If you specify both name+data you specifically want to only delete records matching both fields.

Reply:
Aha, that sounds right.